src/ResourceManagement/ResourceManager/GenericResource/GenericResourcesImpl.cs

```diff
@@ -293,5 +271,29 @@ public async Task<IPagedCollection<IGenericResource>> ListByTagAsync(string reso
                 Manager.Inner.Resources.ListByResourceGroupNextAsync,
                 WrapModel, loadAllPages, cancellationToken);
         }
+
+        internal static string GetApiVersion(string id, IResourceManager resourceManager)
+        {
+            string apiVersionValue;
+            string resourceProvider = ResourceUtils.ResourceProviderFromResourceId(id);
+            string resourceType = ResourceUtils.ResourceTypeForApiVersion(id);
+            string apiVersionKey = resourceProvider + "/" + resourceType;
+            if (cachedApiVersions.ContainsKey(apiVersionKey))
+            {
+                apiVersionValue = cachedApiVersions[apiVersionKey];
+            }
```

Comment:
Use TryGetValue.

Normally, in concurrency, you want one method call to do the job, not 2.